mountpoint-s3/src/prefetch.rs

```diff
@@ -76,22 +71,28 @@ pub struct Prefetcher<Client, Runtime> {
     inner: Arc<PrefetcherInner<Client, Runtime>>,
 }
 
-#[derive(Debug)]
 struct PrefetcherInner<Client, Runtime> {
-    client: Arc<Client>,
+    part_feed: Arc<dyn ObjectPartFeed<Client> + Send + Sync>,
     config: PrefetcherConfig,
     runtime: Runtime,
 }
 
+impl<Client, Runtime> Debug for PrefetcherInner<Client, Runtime> {
```

Comment:
derive(Debug) did not seem to work with `part_feed: Arc<dyn ObjectPartFeed<Client>...>`. Not completely sure why (wrt `client: Arc<Client>`) or whether there is a better solution.

Reply:
It's because the Debug impl for `part_feed: Arc<dyn ObjectPartFeed<Client>...>` has to work for _any_ `ObjectPartFeed` (because the concrete type isn't known), which can only be possible if the `ObjectPartFeed` trait itself guarantees Debug. On the other hand, the `Arc<Client>` version only has to work for the concrete `Client` type(s) that we instantiate it with, which all happen to implement Debug.